src/puppetlabs/pcp/message.clj

```diff
@@ -96,17 +93,21 @@
 
 (s/defn get-json-data :- s/Any
   "Returns the data from the data frame decoded from json"
-  [message :- Message]
-  (let [data (get-data message)
-        decoded (cheshire/parse-string (bytes->string data) true)]
-    decoded))
+  ([message] (get-json-data message "v2.0"))
+  ([message :- Message
+    version]
+   (case version
+     "v1.0" (-> (get-data message)
+                bytes->string
+                (cheshire/parse-string true))
+     (:data message))))
 
 (s/defn get-json-debug :- s/Any
```

Comment:
Actually, that's slightly complicated. If we want to support running the `ping` action against v1 pxp-agent, the broker needs to add a debug chunk to v1 messages. However, we don't currently provide a way to do that, so I'd support dropping it.